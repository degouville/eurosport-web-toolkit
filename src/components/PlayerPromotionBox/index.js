import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import { rgba } from 'polished';
import Button from '../../elements/Button';
import * as breakpoints from '../../breakpoints';
import * as colors from '../../colors';
import { H2, fontAlphaHeadline, fontHelvetica } from '../../typography';
import Check from '../../assets/red-check.svg';

const StyledWrapper = styled.div`
  position: relative;
  ${props =>
    props.pictureUrl &&
    css`
      background: url(${props.pictureUrl}) ${colors.brandPlus2} top right;
      background-size: cover;
    `};
  color: ${colors.coreLightMinus1};
  padding: 40px 20px 45px;
  ${breakpoints.medium(css`
    padding: 54px 34px 86px;
  `)};
  ${breakpoints.medium(css`
    padding: 34px;
  `)};
  &:after {
    content: '';
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background: linear-gradient(270deg, ${rgba(colors.blueZodiac, 0)} 0%, ${colors.blueZodiac} 100%);
    z-index: 0;
  }
`;

const StyledSubContainer = styled.div`
  max-width: 213px;
  ${breakpoints.large(css`
    max-width: 70%;
  `)};
  position: relative;
  z-index: 1;
`;

const StyledTitle = styled(H2)`
  margin-bottom: 30px;
  color: ${colors.athensGray};
  ${fontAlphaHeadline};
`;

const StyledList = styled.ul`
  margin-bottom: 20px;
`;

export const StyledItem = styled.li`
  background: url(${Check}) no-repeat;
  line-height: 19px;
  background-size: 24px 19px;
  padding-left: 35px;
  margin-bottom: 23px;
  ${fontHelvetica};
  font-size: 14px;
`;

const StyledButton = styled(Button)`
  display: inline-block;
  font-size: 11px;
  width: 213px;
`;

const PlayerPromotionBox = ({ className, pictureUrl, title, texts, textButton, linkButton }) => (
  <StyledWrapper className={className} pictureUrl={pictureUrl}>
    <StyledSubContainer>
      <StyledTitle>{title}</StyledTitle>
      <StyledList>
        {texts.map((text, i) => (
          // eslint-disable-next-line react/no-array-index-key
          <StyledItem key={i}>{text}</StyledItem>
        ))}
      </StyledList>
      {linkButton && textButton && (
        <StyledButton href={linkButton} type="primary">
          {textButton}
        </StyledButton>
      )}
    </StyledSubContainer>
  </StyledWrapper>
);

PlayerPromotionBox.propTypes = {
  pictureUrl: PropTypes.string,
  title: PropTypes.string.isRequired,
  texts: PropTypes.arrayOf(PropTypes.string).isRequired,
  textButton: PropTypes.string,
  linkButton: PropTypes.string,
  className: PropTypes.string,
};

PlayerPromotionBox.defaultProps = {
  pictureUrl: null,
  textButton: null,
  linkButton: null,
  className: '',
};

export default PlayerPromotionBox;
