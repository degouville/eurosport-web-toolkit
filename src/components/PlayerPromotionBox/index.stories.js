import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, object } from '@storybook/addon-knobs';
import { PlayerPromotionBox } from '../..';

const playerpromotionbox = storiesOf('Components|PlayerPromotionBox', module);

const texts = [
  'Watch your favourite sports live and on demand',
  'Stream to your device at home and on the go',
  'Grand Slam tennis, cycling, motorsports and much more',
];

playerpromotionbox.add('Player Promotion Box', () => (
  <PlayerPromotionBox
    title={text('title', 'Watch Eurosport')}
    textButton={text('textButton', 'Find out more')}
    texts={object('texts', texts)}
    pictureUrl={text(
      'pictureUrl',
      'https://www.eurosport.fr/watch-eurosport-react/static/media/features-summary-1024.fe17bcd2.jpg'
    )}
    linkButton={text(
      'linkButton',
      'https://www.eurosport.fr/regardez-eurosport-en-direct.shtml?int_campaign=player-france-prospect-allsports-undefined&int_content=home-toolbar-button-watch'
    )}
  />
));
