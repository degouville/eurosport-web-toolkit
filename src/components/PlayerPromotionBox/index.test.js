import React from 'react';
import { shallow } from 'enzyme';
import PlayerPromotionBox, { StyledItem } from '.';

const texts = [
  'Watch your favourite sports live and on demand',
  'Stream to your device at home and on the go',
  'Grand Slam tennis, cycling, motorsports and much more',
];

describe('PlayerPromotionBox', () => {
  it('renders its content', () => {
    expect(
      shallow(
        <PlayerPromotionBox
          title="Watch Eurosport"
          textButton="Find out more"
          texts={texts}
          pictureUrl="https://www.eurosport.fr/watch-eurosport-react/static/media/features-summary-1024.fe17bcd2.jpg"
          linkButton="https://www.eurosport.fr/regardez-eurosport-en-direct.shtml?int_campaign=player-france-prospect-allsports-undefined&int_content=home-toolbar-button-watch"
        />
      )
    ).toMatchSnapshot();
  });

  it('display each list item regarding the texts provided', () => {
    const wrapper = shallow(
      <PlayerPromotionBox
        title="Watch Eurosport"
        textButton="Find out more"
        texts={texts}
        pictureUrl="https://www.eurosport.fr/watch-eurosport-react/static/media/features-summary-1024.fe17bcd2.jpg"
        linkButton="https://www.eurosport.fr/regardez-eurosport-en-direct.shtml?int_campaign=player-france-prospect-allsports-undefined&int_content=home-toolbar-button-watch"
      />
    );
    expect(wrapper.find(StyledItem).length).toEqual(3);
  });
});
