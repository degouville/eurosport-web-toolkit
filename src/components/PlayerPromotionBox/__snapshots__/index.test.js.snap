// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerPromotionBox renders its content 1`] = `
<Styled(div)
  pictureUrl="https://www.eurosport.fr/watch-eurosport-react/static/media/features-summary-1024.fe17bcd2.jpg"
>
  <Styled(div)>
    <Styled(h2)>
      Watch Eurosport
    </Styled(h2)>
    <Styled(ul)>
      <Styled(li)
        key="0"
      >
        Watch your favourite sports live and on demand
      </Styled(li)>
      <Styled(li)
        key="1"
      >
        Stream to your device at home and on the go
      </Styled(li)>
      <Styled(li)
        key="2"
      >
        Grand Slam tennis, cycling, motorsports and much more
      </Styled(li)>
    </Styled(ul)>
    <Styled(ArrowLink)
      href="https://www.eurosport.fr/regardez-eurosport-en-direct.shtml?int_campaign=player-france-prospect-allsports-undefined&int_content=home-toolbar-button-watch"
      type="primary"
    >
      Find out more
    </Styled(ArrowLink)>
  </Styled(div)>
</Styled(div)>
`;
